### PR TITLE
Run PEtab test suite as github action

### PR DESCRIPTION
**Pull request by dweindl**
Run PEtab test suite as github action

### CODE REVIEW

File: tests/petab_test_suite/test_petab_suite.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    gt_simulation_dfs = solution[petabtests.SIMULATION_DFS]
+    tol_chi2 = solution[petabtests.TOL_CHI2]
+    tol_llh = solution[petabtests.TOL_LLH]
+    tol_simulations = solution[petabtests.TOL_SIMULATIONS]
+
+    chi2s_match = petabtests.evaluate_chi2(chi2, solution, tol_chi2)
+    llhs_match = petabtests.evaluate_llh(llh, gt_llh, tol_llh)
+    simulations_match = petabtests.evaluate_simulations(
+        [simulation_df], gt_simulation_dfs, tol_simulations)
+
```
Comment (FFroehlich) [inline tests/petab_test_suite/test_petab_suite.py]:
```suggestion
    for edata in edatas_from_petab(model=model, petab_problem=petab_problem):
        for ip in range(model.np):
            amici.check_finite_difference(x0, model, solver, edata, ip, fields, assert_fun)
```

Comment (dweindl) [reply tests/petab_test_suite/test_petab_suite.py]: Thx

Comment (FFroehlich) [reply tests/petab_test_suite/test_petab_suite.py]: eeeeh probably needs some fine tuning before it actually works :D

File: .github/workflows/test-petab-test-suite.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Benchmark collection
```
Comment (FFroehlich) [inline .github/workflows/test-petab-test-suite.yml]:
```suggestion
name: PEtab testsuite
```

File: .github/workflows/test-petab-test-suite.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      - master
+
+  pull_request:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build:
+    name: Benchmark collection
```
Comment (FFroehlich) [inline .github/workflows/test-petab-test-suite.yml]:
```suggestion
    name: PEtab testsuite
```

File: tests/petab_test_suite/test_petab_suite.py (hunk 1 above)
Comment (dweindl) [reply tests/petab_test_suite/test_petab_suite.py]: yeah, will massage it in ;)

File: .gitignore [hunk 4]
```diff
@@ -38,6 +38,8 @@ models/model_robertson/build/*
 !models/model_calvetti
 models/model_calvetti/build/*
 
+amici_models/*
```
Comment (yannikschaelte) [inline .gitignore]: that is the default I used in pypesto. If that leads to problems, one can rename in amici.petab.petab_import

File: tests/petab_test_suite/test_petab_suite.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    simulation_df = simulation_df.rename(
+        columns={petab.MEASUREMENT: petab.SIMULATION})
+    simulation_df[petab.TIME] = simulation_df[petab.TIME].astype(int)
+    solution = petabtests.load_solution(case)
+    gt_chi2 = solution[petabtests.CHI2]
+    gt_llh = solution[petabtests.LLH]
+    gt_simulation_dfs = solution[petabtests.SIMULATION_DFS]
+    tol_chi2 = solution[petabtests.TOL_CHI2]
+    tol_llh = solution[petabtests.TOL_LLH]
+    tol_simulations = solution[petabtests.TOL_SIMULATIONS]
```
Comment (yannikschaelte) [inline tests/petab_test_suite/test_petab_suite.py]: one can think about the output format of the load_solution if that is not soo nice here, but should work.

File: tests/petab_test_suite/conftest.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+"""Pytest configuration for PEtab test suite"""
+
+from typing import List
+import re
+import sys
+import petabtests
+
+
+def parse_selection(selection_str: str) -> List[int]:
```
Comment (yannikschaelte) [inline tests/petab_test_suite/conftest.py]: is this used?

Comment (dweindl) [reply tests/petab_test_suite/conftest.py]: Yes, allows you to run something like `pytest --petab-cases=7,8,9-12`

File: tests/petab_test_suite/test_petab_suite.py (hunk 5 above)
Comment (dweindl) [reply tests/petab_test_suite/test_petab_suite.py]: Changes welcome. Maybe later...

File: tests/petab_test_suite/conftest.py (hunk 6 above)
Comment (yannikschaelte) [reply tests/petab_test_suite/conftest.py]: so this will allow to only run the tests that amici should support. ok!

File: .gitignore (hunk 4 above)
Comment (dweindl) [reply .gitignore]: Fine for me. Just ignoring, because e.g. gitk otherwise shows hundreds of modified files

**Pull request merged**